### DISCUSSION TITLE
How to use SDXL models from civitai?

### DISCUSSION THREAD

**Discussion by blx0102** [huggingface/diffusers#8892] [Q&A] [answered]
I download some SDXL models from civitai, and used in diffusers like below, but got poor result, the output image appears as if it's covered with a layer of hazy fog, and not clear. What I missed?

diffusers == 0.27.2

Code snippet:
```
base_model = "./models/dreamshaperXL_v21TurboDPMSDE.safetensors"
vae_model_path = "./models/sdxl_vae.safetensors"

device = "cuda"

noise_scheduler = DDIMScheduler(
    num_train_timesteps=1000,
    beta_start=0.00085,
    beta_end=0.012,
    beta_schedule="scaled_linear",
    clip_sample=False,
    set_alpha_to_one=False,
    steps_offset=1,
)
vae = AutoencoderKL.from_single_file(vae_model_path).to(dtype=torch.float16)

pipe = StableDiffusionXLPipeline.from_single_file(
    base_model,
    torch_dtype=torch.float16,
    scheduler=noise_scheduler,
    vae=vae,
    use_safetensors=True,
    add_watermarker=False,
    load_safety_checker=False
).to(device)

images = pipe(
    prompt=prompt,
    negative_prompt=negative_prompt,
    num_images_per_prompt=2,
    guidance_scale=gs,
    width=w,
    height=h,
    num_inference_steps=steps,
).images
```

Results:
![image](https://github.com/user-attachments/assets/49c6ea67-285d-4f17-92dd-9097360e0b24)

**Comment by asomoza** [comment, marked as answer]
HI, that's the vae. Try removing the code for the vae, also try to follow the instructions of the model owner if you want good results:

> Turbo version should be used at CFG scale 2 and with around 4-8 sampling steps. This should work only with DPM++ SDE Karras (NOT 2M). You can use this with LCM sampler, but don't do it unless you need speed vs quality

These are the results I get with that exact same model:

|||
|---|---|
|![demo1](https://github.com/user-attachments/assets/94a7b4de-5446-49c1-825d-46f0e9d8637b)|![issue](https://github.com/user-attachments/assets/727fdc75-cecb-4cd6-9fb8-aecdea05d2d3)|

**Reply by blx0102** [reply]
Thx! That sloved my question.

**Reply by blx0102** [reply]
@asomoza Hi, sorry to disturb, another question. When I using ip-adapter-face-plus-sdxl, the style of output image will highly affected by the style of the face image, for example, if the face image is a realistic photo, the output image will be realistic as well, even with prompt including "anime style".
I see we can set scores to specific part of blocks, like 
```
# To use style+layout blocks
scale = {
    "down": {"block_2": [0.0, 1.0]},
    "up": {"block_0": [0.0, 1.0, 0.0]},
}
```
but the only mentioned blocks are related to style and layout, how can I only use face block?

**Reply by asomoza** [reply]
there's a misunderstanding here, there isn't a "face block", there also isn't a "style" or "layout" block. That's mostly a discovery that those blocks somewhat controls those and in my opinion, it's not 100% accurate because it depends on the training and the type of model.

For example, I found that the first UP attention layer can be used to control more or less the shape of things, like the clothes but it doesn't work all the time, also I found that some other layers sometimes remove the background or affect other things but that's in case by case situation. 

With the face ip adapters, to transform the face you'll need to use a combination of other things to get good results, also it depends on how good or how much do you want to transform the face.

I don't like to use people as examples but it can be avoided here. 

Is this a good result for what you want?

|||
|---|---|
|![man2_face](https://github.com/user-attachments/assets/8b9f2125-53fa-45aa-9de8-bf9bb1fe62de)|![20240721162836](https://github.com/user-attachments/assets/5a5655a0-ae77-4384-9483-054dfa25379b)|

The image is from [unsplash](https://unsplash.com/photos/man-in-black-crew-neck-shirt-qo1SX1HDPnM) by [nartan](https://unsplash.com/@nartan)

If you want more precise solutiosn you'll need to use trained models for this use case. For example [InstandId](https://github.com/InstantID/InstantID) or mix more stuff like in this space [Face to All](https://huggingface.co/spaces/multimodalart/face-to-all)

**Reply by blx0102** [reply]
Alright, thanks for your detailed reply!
